pkg/phlare/context/context.go

```diff
@@ -41,6 +44,17 @@ func Registry(ctx context.Context) prometheus.Registerer {
 	return prometheus.NewRegistry()
 }
 
+func WithLocalBucketClient(ctx context.Context, client phlareobj.Bucket) context.Context {
```

Comment:
Unsure why not? We already pass registry and logger this way and this will make wrap work very similarly: 

https://github.com/grafana/pyroscope/pull/2173/files#diff-e8e445aa08cfb99aca5e6bfc72d3924ca6724b2acf2729678158fee6af0645f8R58

Reply:
I think we should not pass dependencies via the context (which, I believe, registry, logger, and object store client are) – although, this is very opinionated ([`context is for cancelation`](https://dave.cheney.net/2017/01/26/context-is-for-cancelation), [`context isn't for cancelation`](https://dave.cheney.net/2017/08/20/context-isnt-for-cancellation)).

I'd like to highlight one of the principles from https://go.dev/blog/context-and-structs we should follow:
> Contexts should not be stored inside a struct type, but instead passed to each function that needs it.

Which means that the way we use `phlarecontext` is wrong (I partially agree).

On the other hand, if it is already implemented in that way, and it does not make dependency injection unmanageable, it is fine with me. So I see nothing bad in `WithLocalBucketClient`. But I guess that in future we may want to refactor or get rid of the `pkg/phlare/context` package.

Reply:
Ok to not make matters worse, I will pass the stuff around by variable then 🙂